frontend/packages/ux-editor/src/hooks/useLayoutSetIcon.tsx

```diff
@@ -9,17 +9,37 @@ import {
 } from '@studio/icons';
 import type { LayoutSetModel } from 'app-shared/types/api/dto/LayoutSetModel';
 import type { StudioIconCardIconColors } from '@studio/components-legacy/src/components/StudioIconCard/StudioIconCard';
+import type { BpmnTaskType } from 'app-shared/types/BpmnTaskType';
 
-export const useLayoutSetIcon = (
-  layoutSetModel: LayoutSetModel,
-): { icon: ReactElement; iconColor: StudioIconCardIconColors } => {
-  if (layoutSetModel.type == 'subform') return { icon: <ClipboardIcon />, iconColor: 'blue' };
+type IconMetaData = {
+  icon: ReactElement;
+  iconColor: StudioIconCardIconColors;
+};
+
+export const useLayoutSetIcon = (layoutSetModel: LayoutSetModel): IconMetaData => {
+  return (
+    iconByLayoutSetModelType[layoutSetModel.type] ||
+    iconByTaskIdMap[layoutSetModel.task?.id] ||
+    iconByTaskTypeMap[layoutSetModel.task?.type] ||
+    defaultIcon
+  );
+};
+
+const iconByTaskTypeMap: Partial<Record<BpmnTaskType, IconMetaData>> = {
```

Comment:
In a follow-up PR, I will remove the Partial to ensure full type safety. This will help catch missing icons at compile time whenever a new BPMN task type is added, preventing future bugs.
```suggestion
const iconByTaskTypeMap: Record<BpmnTaskType, IconMetaData> = {
```